src/methods/clipping/difference.jl

```diff
@@ -162,6 +162,7 @@ function _difference(
     local polys
     for (i, poly_b) in enumerate(GI.getpolygon(multipoly_b))
         polys = difference(i == 1 ? multipoly_a : GI.MultiPolygon(polys), poly_b; target, fix_multipoly)
+        isempty(polys) && break
```

Comment:
Doesn't this fail if multiply A only intersects the 2nd polygon in multiply B?

(Multiply = multi poly, on mobile)

Reply:
Specifically, considering the case where `multipoly_a` is really just `MultiPolygon(first(GI.getgeom(multipoly_b)))`.  In this case, polys will be empty at i=1, but that short circuiting will ignore the other polygons in poly_b, which are different from multipoly_a.

Reply:
That doesn't matter because we are taking the difference from multipolygon `a`. So once multipolygon `a` is completely covered by any pieces of multipolgon `b`, we have no piece of multipolygon `a` left to take the difference from with the remaining `b` pieces.

Reply:
This is a unique short circuit that applies to `difference`. I 100% agree this type of logic doesn't work with `intersection` or `union`.

Reply:
Gotcha - that  makes sense.